### PR TITLE
Tests/playwright init

### PR DESCRIPTION
**Pull request by worldspawn-web**
What has been added?
- `Playwright` for overall UI testing.
- GitHub workflow.
- Basic test that will make sure that the site loads correctly.

Also removed some lines in App.tsx :)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -26,6 +30,7 @@
     "@storybook/react": "^8.2.7",
     "@storybook/react-vite": "^8.2.7",
     "@storybook/test": "^8.2.7",
+    "@types/node": "^22.1.0",
```
Comment (Canfus) [inline frontend/package.json:38]: Can we use `ts-node` package instead?
Look at size of them:
[@types/node](https://www.npmjs.com/package/@types/node) - 2.08 MB
[ts-node](https://www.npmjs.com/package/ts-node) - 757 kB

File: frontend/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    // },
+    // {
+    //   name: 'Google Chrome',
+    //   use: { ...devices['Desktop Chrome'], channel: 'chrome' },
+    // },
+  ],
+
+  /* Run your local dev server before starting the tests */
+  webServer: {
+    command: "npm run dev",
```
Comment (Canfus) [inline frontend/playwright.config.ts]: We need to use `pnpm` i think

File: frontend/playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    url: "http://localhost:5173",
```
Comment (Canfus) [inline frontend/playwright.config.ts]: Can we use environment variables?

File: frontend/package.json [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
   "type": "module",
   "scripts": {
     "dev": "vite",
     "build": "tsc -b && vite build",
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
     "preview": "vite preview",
     "storybook": "storybook dev -p 6006",
-    "build-storybook": "storybook build"
+    "build-storybook": "storybook build",
+    "tests": "npx playwright test",
```
Comment (Canfus) [inline frontend/package.json]: We need to use pnpm instead.
For example to execute command via pnpm:
`pnpm exec playwright test`

File: frontend/playwright.config.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * See https://playwright.dev/docs/test-configuration.
+ */
+export default defineConfig({
+  testDir: "./tests",
+  /* Run tests in files in parallel */
+  fullyParallel: true,
+  /* Fail the build on CI if you accidentally left test.only in the source code. */
+  forbidOnly: !!process.env.CI,
```
Comment (Canfus) [inline frontend/playwright.config.ts:19]: It is better to use explicit type casting here.
For example:
```ts
Boolean("Its true, cuz i'm not empty!"); // true
Boolean("") // false
```

Comment (worldspawn-web) [reply frontend/playwright.config.ts:19]: Goes by default with Playwright package

File: frontend/playwright.config.ts (hunk 2 above)
Comment (worldspawn-web) [reply frontend/playwright.config.ts]: Done.

File: frontend/package.json (hunk 4 above)
Comment (worldspawn-web) [reply frontend/package.json]: Done.

File: frontend/package.json (hunk 1 above)
Comment (worldspawn-web) [reply frontend/package.json:38]: There is no publicly announced support on Playwright docs, but we can try.

**Comment by worldspawn-web** [comment]
@Canfus something is wrong with `dotenv` on workflow.
I don't really know how to make them "friends" without adding that to package list

Take a look 👀

**Comment by Canfus** [comment]
> @Canfus something is wrong with `dotenv` on workflow. I don't really know how to make them "friends" without adding that to package list
> 
> Take a look 👀

Its cause we don't have `dotenv` in dependencies.
Look at this logs:
image
I thought the `dotenv` package was installed with playwright.

**Comment by Canfus** [comment]
Now we get another error:
image
Its means that `dotenv` package installed correctly, but we need to remove `package-lock.yaml` file from repo.

**Comment by worldspawn-web** [comment]
> > @Canfus something is wrong with `dotenv` on workflow. I don't really know how to make them "friends" without adding that to package list
> > Take a look 👀
> 
> Its cause we don't have `dotenv` in dependencies. Look at this logs:

Well, I misunderstood you before and thought that `dotenv` must not be in dependencies list

**Comment by worldspawn-web** [comment]
FYI

Currently, there will still be `failed` tests after merge. Will be fixed later with GitHub workflow secrets for environmental variables.

**Pull request merged**